### PR TITLE
Fix false positives in `declaration-property-value-no-unknown` for dollar vars and operators.

### PR DESCRIPTION
**Pull request by pamelalozano16**
- Support first-class operators.
- Support SCSS variables as arguments in nested functions.

### CODE REVIEW

File: src/utils/extractFunctionParams.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+"use strict";
+
+module.exports = function (str) {
+  // Regex to match function calls and their parameters
+  const functionCallRegex = /(\w+)\s*\(([^()]*)\)/g;
+  return str.matchAll(functionCallRegex);
```
Comment (kristerkari) [inline src/utils/extractFunctionParams.js:6]: Not sure if this utility is doing the exact same thing, but could we use the existing `parseFunctionArguments` utility?

https://github.com/stylelint-scss/stylelint-scss/blob/master/src/utils/__tests__/parseFunctionArguments.js#L320

File: src/rules/declaration-property-value-no-unknown/index.js [hunk 2]
```diff
@@ -175,6 +178,18 @@ function rule(primary, secondaryOptions) {
...
         if (isIfStatement(value)) return;
+        if (isFunctionCall(value)) {
+          for (const i of extractFunctionParams(value)) {
+            if (i.some(val => isDollarVar(val))) return;
+          }
+        }
+        const operators = findOperators({ string: value });
+        if (operators) {
+          for (const i of operators) {
+            if (value.match(new RegExp("\\s*\\" + i.symbol + "\\s*", "g")))
```
Comment (kristerkari) [inline src/rules/declaration-property-value-no-unknown/index.js:189]: How about:

```js
const unsupportedFunctions = ["clamp", "min", "max", "env"];
const mathOperators = ["+", "/", "-", "*", "%"];
....
```

```js
const operators = findOperators({ string: value }).map(o => o.symbol);

for (const operator of operators) {
  if (mathOperators.includes(operator)) {
    return;
  }
}
```

File: src/utils/extractFunctionParams.js (hunk 1 above)
Comment (pamelalozano16) [reply src/utils/extractFunctionParams.js]: Right! Thanks

**Comment by kristerkari** [comment]
Thanks! Looks really good now. 👍  

I still need to have another look at this tomorrow to review the fixes that you did in be163f4.

File: package.json [hunk 3]
```diff
@@ -13,6 +13,7 @@
     "stylelint": "^16.0.2"
   },
   "dependencies": {
+    "common-tags": "^1.8.2",
```
Comment (kristerkari) [inline package.json:16]: Sorry that I missed this in the previous review round, but this should be in `devDependencies`, not in `dependencies`.

**Review by kristerkari — CHANGES_REQUESTED** [review]

**Comment by pamelalozano16** [comment]
I noticed declaration-property-value-no-unknown overlaps with unit-no-unknown and color-no-invalid-hex. E.g. [Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYAdAdugLgWyQCcBzAS03hQEZCY88BuLXSAGwkMoGIAzHrAL5YQAGhA9SrGADkkeOIhgAPOQAcpAOjABnbaPARME4ghAZMKFOhDKcMTFG3XKAbSyXL17TgCeU1uQ4ALSQRqTEQd5IDkRQ1u4oALoiCdaEAK5STghoCZ4gsGCsREg4pIZBqoQQqjCEvkEAbkis6TBBmBBB6ZgA1p0A7tg5OBkwKRYe1j2kwZ3dfYPDlKNtEx5WBuyEHV3kzQFQQQAWys4oqzAJQpgC+qHGAGIcBDimAFbahvqwqnqI5ny3j8MACmBw52sxTs3ms602wP8gRChmMkRw0SgsUhIGhMFhIEEIAEQA)

What are our thoughts on this?

**Comment by kristerkari** [comment]
> What are our thoughts on this?

It seems to be something that is expected, and mentioned in `declaration-property-value-no-unknown` readme:

---

Screenshot 2024-09-11 at 22 42 27

---

https://github.com/stylelint/stylelint/tree/main/lib/rules/declaration-property-value-no-unknown

**Comment by kristerkari** [comment]
@pamelalozano16 one last thing: could you also run `npm install` to update the `package-lock.json` after moving `common-tags` to dev dependencies?

**Review by kristerkari — APPROVED** [review]
Looks really good now, thanks! 👍

**Pull request merged**